File: app/src/main/java/com/example/narthana/genpass/NewWordDBHelper.java
package com.example.narthana.genpass;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.example.narthana.genpass.WordContract.WordEntry;

/**
 * Created by narthana on 23/10/16.
 */

class NewWordDBHelper extends SQLiteOpenHelper
{
    private static final String DATABASE_NAME = "words.db";
    private static final int DATABASE_VERSION = 1;

    NewWordDBHelper(Context context) { super(context, DATABASE_NAME, null, DATABASE_VERSION); }

    @Override
    public void onCreate(SQLiteDatabase sqLiteDatabase)
    {
        final String CREATE_WORD_TABLE = "CREATE TABLE " + WordEntry.TABLE_NAME + " ("
                + WordEntry._ID + " INTEGER PRIMARY KEY" + ", "
                + WordEntry.COLUMN_WORD + " TEXT UNIQUE NOT NULL" + ", "
                + WordEntry.COLUMN_LEN + " INTEGER NOT NULL" + ");";
        sqLiteDatabase.execSQL(CREATE_WORD_TABLE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1)
    {
        final String DROP_WORD_TABLE = "DROP TABLE " + WordEntry.TABLE_NAME;
        sqLiteDatabase.execSQL(DROP_WORD_TABLE);
        onCreate(sqLiteDatabase);
    }
}

File: app/src/main/java/com/example/android/genpass/data/NewWordDBHelper.kt
package com.example.android.genpass.data

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

/**
 * Created by narthana on 23/10/16.
 */

class NewWordDBHelper(context: Context):
        SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    override fun onCreate(sqLiteDatabase: SQLiteDatabase) {
        sqLiteDatabase.execSQL("""
            CREATE TABLE ${WordContract.WordEntry.TABLE_NAME}
            (${WordContract.WordEntry._ID} INTEGER PRIMARY KEY,
             ${WordContract.WordEntry.COLUMN_WORD} TEXT UNIQUE NOT NULL,
             ${WordContract.WordEntry.COLUMN_LEN} INTEGER NOT NULL);
        """)
    }

    override fun onUpgrade(sqLiteDatabase: SQLiteDatabase, i: Int, i1: Int) {
        sqLiteDatabase.execSQL("DROP TABLE ${WordContract.WordEntry.TABLE_NAME}")
        onCreate(sqLiteDatabase)
    }

    companion object {
        private const val DATABASE_NAME = "words.db"
        private const val DATABASE_VERSION = 1
    }
}
